pkg/controller/migmigration/pod.go

```diff
@@ -377,9 +224,48 @@ func (t *Task) veleroPodCredSecretPropagated(cluster *migapi.MigCluster) (bool,
 				return false, nil
 			}
 		} else {
+
 			return false, nil
 		}
 	}
 
 	return true, nil
 }
+
+// Ensure the stage pods have been deleted.
+func (t *Task) ensureStagePodsTerminated() (bool, error) {
+	client, err := t.getDestinationClient()
+	namespaceList := t.destinationNamespaces()
+	if err != nil {
+		log.Trace(err)
+		return false, err
+	}
+	cLabel, _ := t.Owner.GetCorrelationLabel()
+	podList := corev1.PodList{}
+	for _, ns := range namespaceList {
+		selector := labels.SelectorFromSet(map[string]string{
+			StagePodLabel: t.UID(),
+		})
+		err := client.List(
+			context.TODO(),
+			&k8sclient.ListOptions{
+				Namespace:     ns,
+				LabelSelector: selector,
+			},
+			&podList)
+		if err != nil {
+			return false, err
+		}
+		for _, pod := range podList.Items {
+			if _, found := pod.Labels[cLabel]; !found {
```

Comment:
Stage pods do not have _correlation labels_.

Reply:
Done. Thanks for finding it.